### PR TITLE
INT-6768 - Adding version check for future backwards compatibility work

### PR DESCRIPTION
**Pull request by adam-in-ict**
# Description

Thank you for contributing to a JupiterOne integration!

Please include a summary of the change and which issue is fixed. Please also
include relevant motivation and context. List any dependencies that are required
for this change.

## Summary

Adding a query to a vSphere endpoint to get the current API version.  I went ahead and updated the VM guest identity query to check that we're on a supported version before running it.  A future PR will add similar checks to the namespace endpoints for proper handling of version 6.5.

## Type of change

Please leave any irrelevant options unchecked.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### General Development Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### Integration Development Checklist:

Please leave any irrelevant options unchecked.

- [ ] I have checked for additional permissions required to call any new API
      endpoints, and have documented any additional permissions in
      `jupiterone.md`, where necessary.
- [ ] My changes properly paginate the target service provider's API
- [ ] My changes properly handle rate limiting of the target service provider's
      API
- [ ] My new integration step is instrumented to execute in the correct order
      using `dependsOn`
- [ ] I have referred to the
      [JupiterOne data model](https://github.com/JupiterOne/data-model/tree/main/src/schemas)
      to ensure that any new entities/relationships, and relevant properties,
      match the recommended model for this class of data
- [ ] I have updated the `CHANGELOG.md` file to describe my changes
- [ ] When changes include modifications to existing graph data ingestion, I've
      reviewed all existing managed questions referencing the entities,
      relationships, and their property names, to ensure those questions still
      function with my changes.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -125,6 +143,35 @@ export class APIClient {
...
+      try {
+        const versionResponse: VsphereVersion = await this.request(
+          this.versionEndpoint,
+        );
+        const versionArray = versionResponse.value.version.split('.');
+        this.apiVersion.major = Number(versionArray[0]);
+        this.apiVersion.minor = Number(versionArray[1]);
+        this.apiVersion.patch = Number(versionArray[2]);
+      } catch (err) {
+        this.logger.warn(
```
Comment (adam-in-ict) [inline src/client.ts:157]: I'm on the fence with this one.  Should we catch this and set a default or just bail out completely?

Comment (VDubber) [reply src/client.ts:157]: Is the version endpoint supported by 6.5+?
Does the version endpoint required auth? Could we use it in the validationInvocation method?

File: src/client.ts [hunk 2]
```diff
@@ -19,17 +19,35 @@ import {
   VsphereHost,
   VsphereNamespace,
   VsphereNetwork,
+  VsphereVersion,
   VsphereVm,
   VsphereVmDetails,
 } from './types';
 
+let client: APIClient;
```
Comment (VDubber) [inline src/client.ts:27]: consider moving to bottom of file

**Review by VDubber — DISMISSED** [review]
Minor comments.

File: src/client.ts (hunk 1 above)
Comment (adam-in-ict) [reply src/client.ts:157]: Yes, it is supported by 6.5 and all other versions between that and the current 8.0.0.

Comment (adam-in-ict) [reply src/client.ts:157]: It does require authentication.  We could absolutely add it to the validateInvocation method.

Honestly maybe that *should* be in the validateInvocation call.  And we don't pass validation unless it succeeds.

Comment (VDubber) [reply src/client.ts]: I'd be okay with that!

**Review by VDubber — APPROVED** [review]

**Pull request merged**